### ISSUE TITLE
Issue with immutability of LazyLists in import_videos

### ISSUE THREAD

**Issue by grigorisg9gr** [menpo/menpo#705]
The fact that import_videos is a LazyList of LazyLists while LazyLists are immutable causes issues in the existing framework. 

Specifically, unless I am mistaken, there is no way to know which clip is being currently parsed with import_videos as there is no .path command either in the videos' LazyList or the frames' LazyList. 
For instance in the following code you can reproduce that issue: 
```
for v in mio.import_videos(p_vid):
    try:
        v.path
    except AttributeError:
        print('v of type {} does not have such an attribute.'.format(type(v)))
    try:
        v[0].path
    except AttributeError:
        print('v[0] of type {} does not have such an attribute.'.format(type(v[0])))
    break
```

In, other words, if you call the command import_videos, you have no idea which video is parsed. This is due to the LazyLists being immutable, hence the command https://github.com/menpo/menpo/blob/master/menpo/io/input/base.py#L801 does not add the attribute .path . 

Additionally, this immutability causes other parts not to be exposed, e.g. the fps of a video, even in @JeanKossaifi 's implementation: https://github.com/menpo/menpo/pull/702/files#diff-129a17f8aebeb75602a29355602ef7b5R52 .

**Comment by JeanKossaifi** [comment]
There is a tradeoff to be found to be able to provide a unified interface for images from different sources. In the case of videos, the current implementation returns a LazyList of Menpo images which is more agnostic than just videos and probably shouldn't include information such as fps (in that case if the user wants to access the video information they can instantiate directly a FFMpegVideoReader object).

One solution would be to add an optional `source` attribute to Menpo images to specify where the image was loaded from?

**Comment by patricksnape** [comment]
At the moment that is essentially what the path is. Maybe we could formalise the path?

**Comment by patricksnape** [comment]
This should be fixed more formally by #707
